homeassistant/components/bluetooth/passive_update_processor.py

```diff
@@ -180,19 +259,26 @@ def __init__(
             list[Callable[[PassiveBluetoothDataUpdate[_T] | None], None]],
         ] = {}
         self.update_method = update_method
-        self.entity_names: dict[PassiveBluetoothEntityKey, str | None] = {}
-        self.entity_data: dict[PassiveBluetoothEntityKey, _T] = {}
-        self.entity_descriptions: dict[
-            PassiveBluetoothEntityKey, EntityDescription
-        ] = {}
-        self.devices: dict[str | None, DeviceInfo] = {}
+        data: PassiveBluetoothDataUpdate[_T] = PassiveBluetoothDataUpdate()
+        self.data = data  # This is the combined data from all the updates
+        self.entity_names = data.entity_names
+        self.entity_data = data.entity_data
+        self.entity_descriptions = data.entity_descriptions
+        self.devices = data.devices
         self.last_update_success = True
 
     @property
     def available(self) -> bool:
         """Return if the device is available."""
         return self.coordinator.available and self.last_update_success
 
+    @callback
+    def async_restore_data(self) -> tuple[str, dict[str, Any]] | None:
+        """Restore the data."""
+        if self.restore_key is None:
```

Comment:
If available is false we should return None as well as it means the device is no longer in range and we shouldn't save it

Reply:
Actually it's ok to save it since we will know if the device is in range on next startup and it will be unavailable anyways.  If we throw it away we might loose the state of a sensor that only updates daily